src/core/workload.py

```diff
@@ -0,0 +1,139 @@
+        connect to.
+        """
+        return f"{self.conf_path}/zookeeper-dynamic.properties"
+
+    @property
+    def jaas(self) -> str:
+        """The zookeeper-jaas.cfg filepath.
+
+        Contains internal+external user credentials used in SASL auth.
+        """
+        return f"{self.conf_path}/zookeeper-jaas.cfg"
+
+    @property
+    def jmx_prometheus_javaagent(self) -> str:
+        """The JMX exporter JAR filepath.
+
+        Used for scraping and exposing mBeans of a JMX target.
+        """
+        return f"{self.binaries_path}/lib/jmx_prometheus_javaagent.jar"
+
+    @property
+    def jmx_prometheus_config(self) -> str:
+        """The configuration for the JMX exporter."""
+        return f"{self.conf_path}/jmx_prometheus.yaml"
+
+    @property
+    def server_key(self) -> str:
+        """The private-key for the service to identify itself with for TLS auth."""
+        return f"{self.conf_path}/server.key"
+
+    @property
+    def ca(self) -> str:
+        """The shared cluster CA."""
+        return f"{self.conf_path}/ca.pem"
+
+    @property
+    def certificate(self) -> str:
+        """The certificate for the service to identify itself with for TLS auth."""
+        return f"{self.conf_path}/server.pem"
+
+    @property
+    def truststore(self) -> str:
+        """The Java Truststore containing trusted CAs + certificates."""
+        return f"{self.conf_path}/truststore.jks"
+
+    @property
+    def keystore(self) -> str:
+        """The Java Truststore containing service private-key and signed certificates."""
+        return f"{self.conf_path}/keystore.p12"
+
+
+class WorkloadBase(ABC):
+    """Base interface for common workload operations."""
+
+    paths = ZKPaths()
+
+    @abstractmethod
+    def start(self) -> None:
+        """Starts the workload service."""
+        ...
+
+    @abstractmethod
+    def stop(self) -> None:
+        """Stops the workload service."""
+        ...
+
+    @abstractmethod
+    def restart(self) -> None:
+        """Restarts the workload service."""
+        ...
+
+    @abstractmethod
+    def read(self, path: str) -> list[str]:
+        """Reads a file from the workload.
+
+        Args:
+            path: the full filepath to read from
+
+        Returns:
+            List of string lines from the specified path
+        """
+        ...
+
+    @abstractmethod
+    def write(self, content: str, path: str) -> None:
+        """Writes content to a workload file.
+
+        Args:
+            content: string of content to write
+            path: the full filepath to write to
+        """
+        ...
+
+    @abstractmethod
+    def exec(self, command: list[str], working_dir: str | None = None) -> None:
+        """Runs a command on the workload substrate."""
+        ...
+
+    @abstractmethod
+    def alive(self) -> bool:
+        """Checks that the workload is alive."""
+        ...
+
+    @abstractmethod
+    def healthy(self) -> bool:
```

Comment:
Not sure what the use case for this one is. Neither zookeeper nor kafka charms implement any logic for `workload.healthy`.